model/test_results.go

```diff
@@ -195,6 +208,44 @@ func (t *TestResults) Append(ctx context.Context, results []TestResult) error {
 	return errors.Wrap(bucket.Put(ctx, testResultsCollection, bytes.NewReader(data)), "uploading test results")
 }
 
+func (t *TestResults) appendToFailedTestsSample(ctx context.Context, results []TestResult) error {
+	var update bool
+	for i := 0; i < len(results) && len(t.FailedTestsSample) < failedTestsSampleSize; i++ {
+		if strings.Contains(strings.ToLower(results[i].Status), "fail") {
+			t.FailedTestsSample = append(t.FailedTestsSample, results[i].getDisplayName())
+			update = true
+		}
+	}
+	if !update {
+		return nil
+	}
+
+	if t.ID == "" {
```

Comment:
In what case would the ID not be set? It seems like the ID is generated with `CreateTestResultsRecord` and calling `Append from the gRPC route populates the ID when retrieving it from the DB.